src/pip/setup.py

```diff
@@ -0,0 +1,133 @@
+All rights reserved.
+THIS PROGRAM IS PROVIDED UNDER THE TERMS OF GPL VERSION 3 LICENSE OR
+THIS OSMC PUBLIC LICENSE (OSMC-PL) VERSION 1.2.
+ANY USE, REPRODUCTION OR DISTRIBUTION OF THIS PROGRAM CONSTITUTES
+RECIPIENT'S ACCEPTANCE OF THE OSMC PUBLIC LICENSE OR THE GPL VERSION 3,
+ACCORDING TO RECIPIENTS CHOICE.
+The OpenModelica software and the Open Source Modelica
+Consortium (OSMC) Public License (OSMC-PL) are obtained
+from OSMC, either from the above address,
+from the URLs: http://www.ida.liu.se/projects/OpenModelica or
+http://www.openmodelica.org, and in the OpenModelica distribution.
+GNU version 3 is obtained from: http://www.gnu.org/copyleft/gpl.html.
+This program is distributed WITHOUT ANY WARRANTY; without
+even the implied warranty of  MERCHANTABILITY or FITNESS
+FOR A PARTICULAR PURPOSE, EXCEPT AS EXPRESSLY SET FORTH
+IN THE BY RECIPIENT SELECTED SUBSIDIARY LICENSE CONDITIONS OF OSMC-PL.
+See the full OSMC Public License conditions for more details.
+'''
+
+__author__ = 'Open Source Modelica Consortium (OSMC)'
+__maintainer__ = 'https://openmodelica.org'
+__status__ = 'Production'
+
+
+import os
+import platform
+import shutil
+import sysconfig
+import tempfile
+from distutils.command.build_py import build_py
+from distutils.dir_util import copy_tree
+
+import requests
+from setuptools import setup
+
+
+# overrride build_py, for compiling and copying the dlls
+class my_build_py(build_py):
+  def __init__(self, dist):
+    # check if directory exist and remove
+    topdir = os.path.join(os.getcwd(), 'OMSimulator')
+    if os.path.isdir(topdir):
+      os.rmdir(topdir)
+
+    # recreate it, so that setuptools creates the package directory
+    os.mkdir(topdir)
+
+    build_py.__init__(self, dist)
+
+  def fetch_oms(self):
+    target_dir = os.path.join(self.build_lib, 'OMSimulator')
+
+    # mkpath is a distutils helper to create directories
+    self.mkpath(target_dir)
+
+    # download the zip directory from url
+    if (sysconfig.get_platform() == 'linux-x86_64'):
+      response = requests.get('https://build.openmodelica.org/omsimulator/linux-amd64/OMSimulator-linux-amd64-@OMS_VERSION_STRING@.tar.gz')
+    elif (sysconfig.get_platform() == 'linux-i386'):
+      response = requests.get('https://build.openmodelica.org/omsimulator/linux-i386/OMSimulator-linux-i386-@OMS_VERSION_STRING@.tar.gz')
+    elif (sysconfig.get_platform() == 'linux-arm32'):
+      response = requests.get('https://build.openmodelica.org/omsimulator/linux-arm32/OMSimulator-linux-arm32-@OMS_VERSION_STRING@.tar.gz')
+    elif (sysconfig.get_platform() == 'mingw' and platform.architecture()[0] == '64bit'):
+      response = requests.get('https://build.openmodelica.org/omsimulator/win-mingw64/OMSimulator-mingw64-@OMS_VERSION_STRING@.zip')
+    elif (sysconfig.get_platform() == 'mingw' and platform.architecture()[0] == '32bit'):
+      response = requests.get('https://build.openmodelica.org/omsimulator/win-mingw64/OMSimulator-mingw64-@OMS_VERSION_STRING@.zip')
```

Comment:
the path should be https://build.openmodelica.org/omsimulator/win-mingw32/OMSimulator-mingw32-@OMS_VERSION_STRING@.zip